core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystemImpl.java

```diff
@@ -363,7 +363,12 @@ && getDefaultPrecision(typeName) != RelDataType.PRECISION_NOT_SPECIFIED) {
 
   @Override public RelDataType deriveAvgAggType(RelDataTypeFactory typeFactory,
       RelDataType argumentType) {
-    return argumentType;
+    if (SqlTypeName.INT_TYPES.contains(argumentType.getSqlTypeName())) {
```

Comment:
the type system is actually pluggable. See https://github.com/feldera/feldera/issues/3588 for the right way to do this. We should not change RelDataTypeSystemImpl.